builtin/mainmenu/tab_online.lua

```diff
@@ -56,18 +56,23 @@ local function get_sorted_servers()
 end
 
 -- Persists the selected server in the "address" and "remote_port" settings
+
+local function is_selected_fav(server)
+	local address = core.settings:get("address")
+	local port = tonumber(core.settings:get("remote_port"))
+
+	for _, fav in ipairs(serverlistmgr.get_favorites()) do
+			if address == fav.address and port == fav.port then
+				return true
+			end
+		end
+end
+
 local function set_selected_server(server)
 	local address = server.address
 	local port    = server.port
-	gamedata.serverdescription = server.description
 
-	gamedata.fav = false
-	for _, fav in ipairs(serverlistmgr.get_favorites()) do
-		if address == fav.address and port == fav.port then
-			gamedata.fav = true
-			break
-		end
-	end
+	is_selected_fav()
```

Comment:
```suggestion
```

After this refactor, nothing needs to be done here: `is_selected_fav` is just determined on demand, not eagerly updated here.